app/src/test/java/org/fossasia/openevent/general/RegExTest.kt

```diff
@@ -0,0 +1,18 @@
+package org.fossasia.openevent.general
+
+import org.fossasia.openevent.general.auth.EditProfileFragment
+import org.junit.Assert.assertEquals
+import org.junit.Test
+
+class RegExTest {
+    val frag = EditProfileFragment()
+    @Test
+    fun `protocol removal when loaded`() {
+        assertEquals("abcdefg", frag.loadStringRegEx("https://instagram.com/abcdefg"))
+    }
+
+    @Test
+    fun `anomaly removal when updating`() {
+        assertEquals("helloworld", frag.updateStringRegEx("h@ello@wo ,   rld"))
```

Comment:
Since this is invalid username, it should throw. We are only going to make benign changes like URLs to username conversions and removing of `@` sign and it should be visible to user before they click save

Reply:
but invalid username is confirmed at a different function 😅 that function decides if the username is invalid or not.. this is just a regEx pattern function which takes a string removes any anomaly and return a string.. I will make a new test case for invalid username

Reply:
It should not do this

Reply:
@iamareebjamal so how do we check and change the UI with onChangeListner and remove things when user Clicked on update with a same function 😕 ? this function is being used when updating and also while checking. I added that test if someone changes the pattern and if it leads to deleting of wrong characters .. now I also added another test...

Reply:
I think this solves this conversation